### PR TITLE
Extract stitch request serde code

### PR DESCRIPTION
**Pull request by cgtz**
Previously the code to serialize/deserialize the JSON body of stitch
requests was interspersed throughout different classes. This puts that
code into ambry-api so that projects that depend on ambry can use it
easily.

Also move the operation constants to ambry-api for easier reuse.

### CODE REVIEW

File: ambry-api/src/test/java/com/github/ambry/frontend/StitchRequestSerDeTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  /**
+   * Perform a test of serialization/deserialization with specific inputs.
+   * @param signedChunkIds the list of signed chunk IDs to test with.
+   */
+  private void doSerDeTest(List<String> signedChunkIds) {
+    JSONObject stitchRequestJson = StitchRequestSerDe.toJson(signedChunkIds);
+    assertEquals("Unexpected deserialized signed chunk IDs", signedChunkIds,
+        StitchRequestSerDe.fromJson(stitchRequestJson));
+  }
+}
```
Comment (vgkholla) [inline ambry-api/src/test/java/com/github/ambry/frontend/StitchRequestSerDeTest.java]: minor: newline needed

File: ambry-api/src/test/java/com/github/ambry/frontend/StitchRequestSerDeTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ */
+public class StitchRequestSerDeTest {
+
+  /**
+   * Test {@link StitchRequestSerDe}'s serialization/deserialization methods.
+   * @throws Exception
+   */
+  @Test
+  public void serDeTest() {
+    doSerDeTest(Collections.emptyList());
```
Comment (vgkholla) [inline ambry-api/src/test/java/com/github/ambry/frontend/StitchRequestSerDeTest.java]: do we need a `null` case?

File: ambry-api/src/test/java/com/github/ambry/frontend/StitchRequestSerDeTest.java (hunk 1 above)
Comment (cgtz) [reply ambry-api/src/test/java/com/github/ambry/frontend/StitchRequestSerDeTest.java]: added

File: ambry-api/src/test/java/com/github/ambry/frontend/StitchRequestSerDeTest.java (hunk 2 above)
Comment (cgtz) [reply ambry-api/src/test/java/com/github/ambry/frontend/StitchRequestSerDeTest.java]: added

**Comment by vgkholla** [comment]
Will merge once the travis build succeeds

**Pull request merged**